__tests__/app.test.js

```diff
@@ -97,6 +97,27 @@ describe("/api/articles", () => {
     });
 });
 
+describe("/api/articles?topic=", () => {
+    test("GET:200 responds with a filtered array of articles whose topic matches the query parameter", () => {
+        return request(app)
+            .get("/api/articles?topic=mitch")
+            .expect(200)
+            .then(({ body }) => {
+                const { articles } = body;
+                expect(articles).toHaveLength(12);
+                articles.forEach(({ topic }) => expect(topic).toBe("mitch"));
+            });
+    });
+    test("GET:404 responds a suitable error when given a valid but non-existent topic", () => {
+        return request(app)
+            .get("/api/articles?topic=cooking")
+            .expect(404)
+            .then(({ body }) => {
+                expect(body.msg).toBe("slug not found");
+            });
+    });
+});
+
```

Comment:
We should test to make sure if a topic is in the database but has no comments that we get an empty array back, i think `paper` is a topic that this is true for